ingestion/tests/unit/topology/pipeline/test_airbyte.py

```diff
@@ -0,0 +1,138 @@
+            "authProvider": "no-auth",
+        }
+    },
+}
+
+
+EXPECTED_ARIBYTE_DETAILS = AirbytePipelineDetails(
+    workspace=mock_data["workspace"][0], connection=mock_data["connection"][0]
+)
+
+MOCK_CONNECTION_URI_PATH = "/workspaces/af5680ec-2687-4fe0-bd55-5ad5f020a603/connections/a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f"
+MOCK_LOG_URL = f"http://localhost:1234{MOCK_CONNECTION_URI_PATH}"
+
+
+EXPECTED_PIPELINE_STATUS = [
+    OMetaPipelineStatus(
+        pipeline_fqn="airbyte_source.a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+        pipeline_status=PipelineStatus(
+            executionStatus=StatusType.Pending.value,
+            taskStatus=[
+                TaskStatus(
+                    name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+                    executionStatus=StatusType.Pending.value,
+                    startTime=1655482894,
+                    endTime=None,
+                    logLink=f"{MOCK_LOG_URL}/status",
+                )
+            ],
+            executionDate=1655482894,
+        ),
+    ),
+    OMetaPipelineStatus(
+        pipeline_fqn="airbyte_source.a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+        pipeline_status=PipelineStatus(
+            executionStatus=StatusType.Successful.value,
+            taskStatus=[
+                TaskStatus(
+                    name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+                    executionStatus=StatusType.Successful.value,
+                    startTime=1655393914,
+                    endTime=1655394054,
+                    logLink=f"{MOCK_LOG_URL}/status",
+                )
+            ],
+            executionDate=1655393914,
+        ),
+    ),
+]
+
+
+EXPECTED_CREATED_PIPELINES = CreatePipelineRequest(
+    name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+    displayName="MSSQL <> Postgres",
+    description="",
+    pipelineUrl=MOCK_CONNECTION_URI_PATH,
+    tasks=[
+        Task(
+            name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+            displayName="MSSQL <> Postgres",
+            description="",
+            taskUrl=f"{MOCK_CONNECTION_URI_PATH}/status",
+        )
+    ],
+    service=EntityReference(id=uuid.uuid4(), type="pipelineService"),
+)
+
+
+class AirbyteUnitTest(TestCase):
+    def __init__(self, methodName: str = ...) -> None:
+        super().__init__(methodName)
+        self.init_workflow()
+
+    @patch("metadata.ingestion.source.pipeline.pipeline_service.test_connection")
+    @patch("metadata.ingestion.source.pipeline.airbyte.AirbyteClient")
+    def init_workflow(self, airbyte_client, test_connection):
+        test_connection.return_value = False
+        self.client = airbyte_client.return_value
+        self.client.list_jobs.return_value = mock_data.get("jobs")
+        self.client.list_workspaces.return_value = mock_data.get("workspace")
+        self.client.list_connections.return_value = mock_data.get("connection")
+        self.workflow = Workflow.create(mock_airbyte_config)
+        self.workflow.execute()
```

Comment:
sorry to be picky, but this being the first test involving the topologies, it might be interesting for us to get it top notch so that we can use it as a reference

Reply:
So I'm executing the workflow just for setting up the context, but yes I would agree to your point and see if that can be eliminated

Reply:
oh I see. Maybe the context we can mock as well. IMO we would just actually need the Source, not the whole Workflow.

E.g., we can manually create our `AirbyteSource` (as we have other tests managing the proper workflow parsing etc), and at each test, manually specify how the context needs to look like based on the previous step `Create` assert